ext2/ext2.c

```diff
@@ -74,29 +75,32 @@ int ext2_destroy(ext2_t *fs, id_t id)
 }
 
 
-int ext2_lookup(ext2_t *fs, id_t id, const char *name, uint8_t len, oid_t *res, oid_t *dev)
+int ext2_lookup(ext2_t *fs, id_t id, const char *name, size_t len, oid_t *res, oid_t *dev)
 {
 	ext2_obj_t *dir, *obj = NULL;
-	uint8_t i, j;
+	size_t i, j;
 	int err;
 
 	res->port = fs->port;
-	if (!len || (name == NULL))
+	if ((len == 0) || (name == NULL)) {
 		return -EINVAL;
+	}
 
-	if ((dir = ext2_obj_get(fs, id)) == NULL)
+	if ((dir = ext2_obj_get(fs, id)) == NULL) {
 		return -ENOENT;
+	}
 
+	mutexLock(dir->lock);
```

Comment:
Hmm, now we might take `dir->lock` once, but unlock it many times in for below, don't we?

Reply:
This works because the object pointed to by `dir` changes with each iteration. I could rewrite this loop to make it more readable.

Reply:
It's ok, let's keep style/structure improvements as a separate task for later